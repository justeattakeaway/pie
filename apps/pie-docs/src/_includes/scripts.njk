{# All JS modules to go here #}
<script defer src="{{ '/assets/js/f-cookie-banner.js' | url }}" type="module"></script>
<script defer src="{{ '/assets/js/accordions.js' | url }}" type="module"></script>
<script defer src="{{ '/assets/js/toggled-content.js' | url }}" type="module"></script>
<script defer src="{{ '/assets/js/prism.js' | url }}" type="module"></script>
<script defer src="{{ '/assets/js/awesome.js' | url }}" type="module"></script>
<script defer src="{{ '/assets/js/contents.js' | url }}" type="module"></script>
<script defer src="{{ '/assets/js/scroll-to-top.js' | url }}" type="module"></script>
<script defer src="{{ '/assets/js/show-more.js' | url }}" type="module"></script>
<script defer src="{{ '/assets/js/docsearch.js' | url }}" type="module"></script>
<script defer src="{{ '/assets/js/search.js' | url }}" type="module"></script>
