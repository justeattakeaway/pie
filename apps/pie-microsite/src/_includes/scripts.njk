{# All JS modules to go here #}
<script defer src="/assets/js/accordions.js" type="module"></script>
