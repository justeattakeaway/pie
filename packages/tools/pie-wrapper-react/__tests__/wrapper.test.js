import { addReactWrapper, loadCustomElementsFile } from '../scripts/add-react-wrapper';

describe('React Wrapper', () => {
    let mockExample;

    beforeAll(() => {
        mockExample = loadCustomElementsFile('__tests__/mocks');
    });

    it('should be added from mock custom elements JSON', () => {
        const wrapper = addReactWrapper(mockExample);

        expect(wrapper).toMatchSnapshot();
    });

    it('should only be added if the component contains custom elements', () => {
        mockExample.modules.forEach((m) => {
            m.declarations[0].customElement = false;
        });

        const wrapper = addReactWrapper(mockExample);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.length).toBe(0);
    });

    it('should leave the events object empty if the component contains no custom events', () => {
        mockExample.modules.forEach((m) => {
            m.declarations[0].customElement = true;
            delete m.declarations[0].events;
        });

        const wrapper = addReactWrapper(mockExample);

        const result = 'events: {}';

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.includes(result)).toBe(true);
    });

    it('should be named the same as the component itself', () => {
        mockExample.modules.forEach((m) => {
            m.declarations[0].name = 'ButtonComponent';
        });

        const wrapper = addReactWrapper(mockExample, 'pie-wrapper-react');

        const result = 'export const ButtonComponent';

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.includes(result)).toBe(true);
    });
});
