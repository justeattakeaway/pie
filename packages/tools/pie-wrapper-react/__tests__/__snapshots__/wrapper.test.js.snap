// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Wrapper should be added from mock custom elements JSON 1`] = `
"
import * as React from 'react';
import { createComponent, EventName } from '@lit/react';
import { MockComponent as MockComponentReact } from './index';

export * from './defs';

export const MockComponent = createComponent({
    displayName: 'MockComponent',
    elementClass: MockComponentReact,
    react: React,
    tagName: 'mock-component',
    events: {
        onCustomEvent: 'custom-event' as EventName<CustomEvent>,
        onAnotherCustomEvent: 'another-custom-event' as EventName<CustomEvent>,
    },
});
"
`;

exports[`React Wrapper should be named the same as the component itself 1`] = `
"
import * as React from 'react';
import { createComponent } from '@lit/react';
import { ButtonComponent as ButtonComponentReact } from './index';

export * from './defs';

export const ButtonComponent = createComponent({
    displayName: 'ButtonComponent',
    elementClass: ButtonComponentReact,
    react: React,
    tagName: 'mock-component',
    events: {},
});
"
`;

exports[`React Wrapper should leave the events object empty if the component contains no custom events 1`] = `
"
import * as React from 'react';
import { createComponent } from '@lit/react';
import { MockComponent as MockComponentReact } from './index';

export * from './defs';

export const MockComponent = createComponent({
    displayName: 'MockComponent',
    elementClass: MockComponentReact,
    react: React,
    tagName: 'mock-component',
    events: {},
});
"
`;

exports[`React Wrapper should only be added if the component contains custom elements 1`] = `""`;
